synapse/util/threepids.py

```diff
@@ -48,3 +49,32 @@ def check_3pid_allowed(hs, medium, address):
         return True
 
     return False
+
+
+def canonicalise_email(address: str) -> str:
+    """'Canonicalise' email address
+    Case folding of local part of email address and lowercase domain part
+    See MSC2265, https://github.com/matrix-org/matrix-doc/pull/2265
+
+    Args:
+        address: email address to be canonicalised
+    Returns:
+        The canonical form of the email address
+    Raises:
+        ValueError if the address could not be parsed.
+    """
+
+    address = address.strip()
+    # Validate address
+    # See https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-11340
+    parsedAddress = email.utils.parseaddr(address)[1]
```

Comment:
what is the object of this? `email.utils.parseaddr`  is not intended as a way to validate addresses, so I think it is redundant.

Reply:
You use this check in sydent: https://github.com/matrix-org/sydent/commit/4e1cfff53429c49c87d5c457a18ed435520044fc
But it does not find a missing `@`

Reply:
I have no real idea why sydent does that; I wouldn't recommend cargo-culting it. Suggest removing it.